### ISSUE TITLE
AuthenticationJsonWebToken should have a public constructor

### ISSUE THREAD

**Issue by destan** [auth0/auth0-spring-security-api#45]
I think AuthenticationJsonWebToken should have a public constructor otherwise it's impossible to write a custom authentication provider by extending JwtAuthenticationProvider. Because in auth. provider we can't call com.auth0.spring.security.api.authentication.AuthenticationJsonWebToken#setAuthenticated instead we're encouraged to create a new instance of AuthenticationJsonWebToken.

It's very convenient to create a custom authentication provider by extending JwtAuthenticationProvider for example to have a custom revocation or black list logic.

What are your comments about it?

**Comment by rhanton** [comment]
I completely agree. We have the same needs of manually populating an AuthenticationJsonWebToken ourselves (to feed to the JwtAuthenticationProvider).
